Sources/Core/AWSClientRuntime/EventStream/AWSMessageSigner.swift

```diff
@@ -62,10 +65,22 @@ extension AWSEventStream {
         /// - Returns: Signed `Message` with `:chunk-signature` & `:date` headers
         public func signEmpty() async throws -> ClientRuntime.EventStream.Message {
             let signingConfig = try await self.signingConfig()
-            let signingResult = try await AWSSigV4Signer.signEvent(payload: .init(),
+            let sigv4signer = try await self.getSigV4Signer()
+            let signingResult = try await sigv4signer.signEvent(payload: .init(),
                                                                    previousSignature: previousSignature,
                                                                    signingConfig: signingConfig)
             return signingResult.output
         }
+
+        private func getSigV4Signer() async throws -> AWSSigV4Signer {
+            // Fetch resolved signer from selected auth scheme in middleware context
+            let signer = try await self.signer()
+            // For now, only AWSSigV4Signer is used for eventstream message signing
+            let sigv4signer = signer as? AWSClientRuntime.AWSSigV4Signer
```

Comment:
let's refactor this code now assuming that another signer might be added in the future. Could do a resolveMessageSigner() function that returns SigV4 for example

Reply:
Refactored signer structure to accept any custom signer for eventstream message signing.